flashbax/vault/vault.py

```diff
@@ -452,20 +459,48 @@ def read(
 
         # By default we read the entire vault
         if timesteps is None and percentiles is None:
-            read_interval = (0, self.vault_index)
+            # If the vault is bigger than the original fbx buffer, we read the whole
+            # vault and return a *full* buffer state.
+            # Otherwise, we still read the whole vault but return a buffer state of
+            # size equal to the original fbx buffer
+            read_interval = (
+                0,
+                max(
+                    (
+                        self._buffer_time_axis_len,
+                        self.vault_index,
+                    )
+                ),
+            )
+            read_buffer_is_full = self.vault_index >= self._buffer_time_axis_len
+            read_buffer_index = self.vault_index
```

Comment:
Does this make sense in every case? If the vault buffer is larger than the buffer time axis then a full buffer is returned, wouldn't the read buffer index then need to be set to zero?